### PR TITLE
sample for dynamic SDCA binary

### PR DESCRIPTION
**Pull request by sfilipi**
Addresses part of #1209 by adding a sample for the dynamic SDCA binary classification.

### CODE REVIEW

File: docs/samples/Microsoft.ML.Samples/Dynamic/SDCA.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                    {
+                        new TextLoader.Column("Sentiment", DataKind.BL, 0),
+                        new TextLoader.Column("SentimentText", DataKind.Text, 1)
+                    }
+                });
+            
+            // Read the data
+            var data = reader.Read(dataFile);
+
+            // Step2: Pipeline. Featurize the text column through the FeaturizeText.
```
Comment (singlis) [inline docs/samples/Microsoft.ML.Samples/Dynamic/SDCA.cs]: minor nit pick in consistency. Above we say "Step one:" then here we say "Step2:" these should be consistent. Also the sentence starts with Pipeline. Is Pipeline meant to be a header?

File: docs/samples/Microsoft.ML.Samples/Dynamic/SDCA.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                        new TextLoader.Column("Sentiment", DataKind.BL, 0),
+                        new TextLoader.Column("SentimentText", DataKind.Text, 1)
+                    }
+                });
+            
+            // Read the data
+            var data = reader.Read(dataFile);
+
+            // Step2: Pipeline. Featurize the text column through the FeaturizeText. 
+            // Than append a binary classifier, setting the "Label" column, as the column of the dataset, and
```
Comment (singlis) [inline docs/samples/Microsoft.ML.Samples/Dynamic/SDCA.cs]: Than should be Then. I also think some commas need to be moved (i.e. Then append a binary classifier setting the "Label" column as the column of the dataset and the "Features" column, produced by FeaturizeText, as the features column.)

File: docs/samples/Microsoft.ML.Samples/Dynamic/SDCA.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                    .Append(mlContext.BinaryClassification.Trainers.StochasticDualCoordinateAscent(label: "Sentiment", features: "Features", l2Const: 0.001f));
+
+            // Step 3: Run Cross-Validation on this pipeline, and dataFile.
+            var cvResults = mlContext.BinaryClassification.CrossValidate(data, pipeline, labelColumn: "Sentiment");
+
+            var accuracies = cvResults.Select(r => r.metrics.Accuracy);
+            Console.WriteLine(accuracies.Average());
+
+            // If we wanted to specify more advanced parameters for the algorithm, 
+            // we could do so tweaking the AdvancedSetting.
```
Comment (singlis) [inline docs/samples/Microsoft.ML.Samples/Dynamic/SDCA.cs]: I think by is missing - ...do so by tweaking the ...

File: docs/samples/Microsoft.ML.Samples/Program.cs [hunk 4]
```diff
@@ -1,15 +1,15 @@
...
 
 namespace Microsoft.ML.Samples
 {
     internal static class Program
     {
         static void Main(string[] args)
         {
-            Trainers.SdcaRegression();
-            Transformers.ConcatEstimator();
+            TrainerSamples.SDCA_BinaryClassificaiton();
```
Comment (singlis) [inline docs/samples/Microsoft.ML.Samples/Program.cs]: Classification is misspelled.

File: docs/samples/Microsoft.ML.Samples/Dynamic/SDCA.cs [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+                                  (label: "Sentiment",
+                                   features: "Features",
+                                   advancedSettings: s=>
+                                       {
+                                           s.ConvergenceTolerance = 0.01f;   // The learning rate for adjusting bias from being regularized
+                                           s.NumThreads = 2;            // Degree of lock-free parallelism 
+                                       })
+                                   );
+
+            // Run Cross-Validation on this second pipeline, and dataFile.
```
Comment (singlis) [inline docs/samples/Microsoft.ML.Samples/Dynamic/SDCA.cs]: remove comma -- pipeline and dataFile.

File: src/Microsoft.ML.SamplesUtils/SamplesDatasetUtils.cs [hunk 6]
```diff
@@ -10,16 +10,23 @@ namespace Microsoft.ML.SamplesUtils
...
+        /// </summary>
         public static string DownloadHousingRegressionDataset()
-        {
-            string baseGitPath = "https://raw.githubusercontent.com/dotnet/machinelearning/024bd4452e1d3660214c757237a19d6123f951ca/test/data/";
+        => Download("https://raw.githubusercontent.com/dotnet/machinelearning/024bd4452e1d3660214c757237a19d6123f951ca/test/data/housing.txt", "housing.txt");
 
-            // Downloading a regression dataset from github.com/dotnet/machinelearning
-            string dataFile = "housing.txt";
+        /// <summary>
+        /// Downloads the adult dataset from the ML.NEt repo
```
Comment (singlis) [inline src/Microsoft.ML.SamplesUtils/SamplesDatasetUtils.cs]: ML.Net

File: src/Microsoft.ML.StandardLearners/Standard/SdcaCatalog.cs [hunk 7]
```diff
@@ -60,6 +60,18 @@ public static class SdcaBinaryClassificationExtensions
         /// <param name="l1Threshold">The L1 regularization hyperparameter. Higher values will tend to lead to more sparse model.</param>
         /// <param name="maxIterations">The maximum number of passes to perform over the data.</param>
         /// <param name="advancedSettings">A delegate to set more settings.</param>
+        /// <example>
```
Comment (singlis) [inline src/Microsoft.ML.StandardLearners/Standard/SdcaCatalog.cs]: Is this duplicated with the section below? It looks like its grabbing range 48-49 which does not grab much.

Comment (sfilipi) [reply src/Microsoft.ML.StandardLearners/Standard/SdcaCatalog.cs]: This emphasizes the exact line that SDCA gets called in the sample. 
The second one is the full example.

---
In reply to: [227956416](https://github.com/dotnet/machinelearning/pull/1361#discussion_r227956416) [](ancestors = 227956416)

**Comment by sfilipi** [comment]
thanks for taking a look @Ivanidzo4ka and @singlis

**Pull request merged**